### ISSUE TITLE
Questions about paper

### ISSUE THREAD

**Issue by jaeseokbyun** [KevinLight831/CTP#1]
Thanks for sharing great work and dataset!
I have two questions about paper.

First of all, I think that authors mainly follow the losses and architecture of ALBEF. But, CTP do not use the ITM loss as the loss function. Is there any reason? Moreover, do you have any experimental results when ITM loss is added?

Secondly, what is the main difference between "cross-modal retrieval" vs "Multi-modal retrieval" and what is the reason of the experiment about multi-modal retrieval? I guess that the latter use the [CLS] representation of multi-modal encoder and calculate the zero-shot clustering (same class). Similar to the first question, if the ITM loss is added, I wonder about the results of cross-modal retrieval and multi-modal retrieval.

**Comment by KevinLight831** [comment]
Thank you for your question. I would like to take this opportunity to explain in detail our motivations and considerations behind our design.

Q1:

ITM's classifier is commonly utilized as a post-processing step in retrieval. It involves retrieving the top-K samples based on embedding alignment and then scoring them using the classifier for reranking. However, we chose not to adopt ITM for the following three reasons:

1. Our research focuses on developing a general continual pretraining approach, rather than a specific multimodal pretraining model. Our primary objective is to investigate **the modality alignment and fusion capabilities in continual pretraining** and identify effective continual pretraining methods.  Similar to CLIP, modality alignment is achieved through ITC, and embedding alignment is evaluated through cross-modal retrieval, which has met our research objectives about modality alignment. Introducing additional rerank steps through ITM not only makes the settings more complex but also impacts the cross-modal retrieval indicators, making it difficult to accurately evaluate and analyze the ability of various continual learning methods in maintaining modal alignment. 
2. The introduction of ITM will complicate the re-implementation of baseline methods. To re-implement the continual learning method in classification, we move the alignment of 'feature - label' in classification to the alignment of 'image - text' in VLP, Thus, we only need to re-implement each method by changing ITC. Introducing ITM as an additional classifier would lead to more ambiguous re-implementations. Using LWF as an example, on the basis that ITC has already been reimplemented, does ITM align the correct one-hot label or the logit of the previous task model, or both?
3. In addition, ITM and its rerank post-processing also increase training and evaluation time but rerank is not the focus of our study.

To sum up, we excluded the use of ITM in the baseline model and did not conduct any experiments with ITM loss.

Q2:

The operation of multi-modal retrieval is consistent with what you said. The purpose of multi-modal retrieval is to intuitively evaluate the zero-shot clustering ability of multi-modal fusion features (e.g. image+text) without the influence of fine-tuning on downstream tasks. If the clustering effect of multimodal embedding is better, it indicates that the overall multimodal fusion is more effective in achieving compactness within each class and differentiation between classes.
 
This task and evaluation method is mostly used for commodity-related tasks. For more materials about similar tasks or related evaluations, please refer to [1][2][3][4][5].

[1] Product1M: Towards Weakly Supervised Instance-Level Product Retrieval via Cross-Modal Pretraining
[2] M5Product: Self-harmonized Contrastive Learning for E-commercial Multi-modal Pretraining
[3] CommerceMM: Large-Scale Commerce MultiModal Representation Learning with Omni Retrieval
[4] VIMER-UMS : https://wenxin.baidu.com/wenxin/modelbasedetail/vimer_ums
[5] TIANCHI: https://tianchi.aliyun.com/dataset/75730

**Comment by jaeseokbyun** [comment]
Thanks for the clarification.

Given the minimal performance difference between seqF and continual-based methods in "multi-modal retrieval", it raises the question of whether the multi-modal encoder, when employing the ITM loss, is resilient to catastrophic forgetting in the original evaluation setting in ALBEF ("cross-modal retrieval").

**Comment by KevinLight831** [comment]
Sorry for keeping you waiting, as I needed to run some experiments. I introduced ITM into the training process (which adds additional ITM classification parameters and consumes more forward memory), but it is not used for the rerank operation. (As mentioned before, the two-stage post-processing of rerank can blur the measurement of cross-modal alignment ability.) In other words, ITM and related parameters are only used for training and not for the final model inference. From this point of view, the final models of all baseline methods still have the same number of parameters, making it fair for comparison. The experimental results are as follows:

  | cross-modal retrieval（Rm） | multi-modal retrieval（mAP@1）
-- | -- | --
seqF | 56.31 | 61.93
seqF + ITM | 56.32 | 51.64
JointT | 79.93 | 64.62
JointT  + ITM | 79.17 | 59.54
CTP | 64.39 | 62.04
CTP + ITM | —— | ——

Due to the fact that ’CTP+ITM‘ has exceeded the memory limit of 40G A100, we were unable to obtain results for this experiment. All logs and modified clip_pretrain.py files used in this experiment can be found at this [link](https://drive.google.com/drive/folders/1B-gPBSu6l-0oPGCsnC_P2yaCuTzjF1q4?usp=sharing).

The results indicate that adding ITM does not enhance cross-modal alignment ability. On the contrary, it significantly reduces the multi-modal retrieval capability. This may be due to the introduction of the training loss of ITM, which may lead to insufficient training of the original MLM.

In regards to the issue of why there isn't a significant difference in various methods for multimodal retrieval, we have some speculations in section 5.2 of the original paper titled 'Cross-modal vs. Multi-modal'.:

1. The redundancy and complementarity of multi-modal information.
2. the predefined and well-initialized token space in BERT.

Besides, I have another uncertain guess regarding the evaluation method in multimodal retrieval. The current approach of evaluating multiple-to-multiple retrieval may not be as precise in homogeneous product domains as it is in cross-modal retrieval, which uses a one-to-one retrieval approach. This could explain why there isn't a significant difference in multimodal retrieval metrics between different methods. Perhaps a more refined metric is needed to accurately measure multimodal retrieval ?

I understand that it may be difficult for some to accept the similarity in multimodal retrieval results between different continual learning methods. However, as researchers, it is important to convey the information that the data presents, rather than sugarcoat it to avoid controversy. To support our findings, I have provided the experimental logs.

It is also important to note that our goal is to study the changes in cross-modal alignment and multimodal fusion capabilities of pre-trained models under continual pretraining, rather than solely focusing on improving retrieval accuracy like most offline-trained VLP models. Retrieval is merely a tool to evaluate the alignment and fusion capability of continually pre-trained models. Therefore, our research objective does not involve any post-processing or two-stage operations to make the retrieval metrics look better.

If you feel that there are more issues worth exploring, downloading the dataset and code to conduct your experiments is a good option. Please feel free to contact me promptly if there are any bugs in this code repository that I have overlooked. 

Best wishes.

**Comment by jaeseokbyun** [comment]
Thanks for the detailed and precise explanation and additional experiments!
